ai-integrations/haystack.ipynb

```diff
@@ -180,16 +191,16 @@
     "\n",
     "# Setting up a retrieval-augmented generation (RAG) pipeline for generating responses.\n",
     "rag_pipeline = Pipeline()\n",
-    "rag_pipeline.add_component(\"text_embedder\", OpenAITextEmbedder())\n",
+    "rag_pipeline.add_component(\"text_embedder\", VoyageTextEmbedder())\n",
     "\n",
     "# Adding a component for retrieving related documents from MongoDB based on the query embedding.\n",
     "rag_pipeline.add_component(instance=MongoDBAtlasEmbeddingRetriever(document_store=document_store,top_k=15), name=\"retriever\")\n",
     "\n",
     "# Building prompts based on retrieved documents to be used for generating responses.\n",
-    "rag_pipeline.add_component(instance=PromptBuilder(template=prompt_template), name=\"prompt_builder\")\n",
+    "rag_pipeline.add_component(\"prompt_builder\", PromptBuilder(template=prompt_template))\n",
     "\n",
     "# Adding a language model generator to produce the final text output.\n",
-    "rag_pipeline.add_component(instance=OpenAIGenerator(), name=\"llm\")\n",
+    "rag_pipeline.add_component(\"llm\", OpenAIGenerator())\n",
```

Comment:
I can't comment on the lines below, but I'm getting the following warning when I execute this block:

`PromptBuilder has 2 prompt variables, but 'required_variables' is not set. By default, all prompt variables are treated as optional, which may lead to unintended behavior in multi-branch pipelines. To avoid unexpected execution, ensure that variables intended to be required are explicitly set in 'required_variables'.`

I do also get the result `Mark lives in Berlin.`, but I wonder if we want to supply what `PromptBuilder` is expecting so we don't get this warning.